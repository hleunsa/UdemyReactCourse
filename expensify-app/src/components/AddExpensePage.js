import React from 'react';

import ExpenseForm from './ExpenseForm'

const AddExpensePage = () => (
    <div>
        This is from expense component
        <ExpenseForm/>
    </div>
);

export default AddExpensePage;