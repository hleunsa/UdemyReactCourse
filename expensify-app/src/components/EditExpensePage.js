import React from 'react';

const EditExpensePage = () => (
    <div>
        This is from edit component
    </div>
);

export default EditExpensePage;