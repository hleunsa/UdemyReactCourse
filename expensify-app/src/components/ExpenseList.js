import React from 'react';
import { connect } from 'react-redux';

import ExpenseListitem  from './ExpenseListItem'
import { getVisibleExpenses }  from '../selectors/expenses'

const ExpenseList = (props) => (
    <div>
        <p> Expense List </p>
        {props.expenses.map(expense => <ExpenseListitem key={expense.id} {...expense}/>)}
    </div>
);

const mapStateToProps = ({expenses, filters}) => {
    const visibleExpenses = getVisibleExpenses(expenses, filters);
    return {
        expenses: visibleExpenses
    };
};

export default connect(mapStateToProps)(ExpenseList);