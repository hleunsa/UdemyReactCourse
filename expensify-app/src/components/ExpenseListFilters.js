import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';

const ExpenseListFilter = ({filters, dispatch}) => (
    <div>
        <input type='text' value={filters.text} onChange={(e) => {
            dispatch(setTextFilter(e.target.value));
        }}/>
    </div>
);



const mapStateToProps = ({filters}) => ({ filters });

export default connect(mapStateToProps) (ExpenseListFilter);