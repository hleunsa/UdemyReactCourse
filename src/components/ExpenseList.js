import React from 'react';
import { connect } from 'react-redux';

import ExpenseListitem  from './ExpenseListItem';
import { getVisibleExpenses }  from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item list-item--message">
                        <span>No expenses</span>
                    </div>
                ) : (
                    props.expenses.map(expense => <ExpenseListitem key={expense.id} {...expense}/>)
                )
            }
        </div>

    </div>
);

const mapStateToProps = ({expenses, filters}) => {
    const visibleExpenses = getVisibleExpenses(expenses, filters);
    return {
        expenses: visibleExpenses
    };
};

export default connect(mapStateToProps)(ExpenseList);