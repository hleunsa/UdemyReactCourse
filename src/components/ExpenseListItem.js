import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, note, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}><h3> {description} </h3></Link>
        <p> Amount : {amount} </p>
        <p> Date : {createdAt} </p>
        <p> Note : {note} </p>
    </div>
);

export default ExpenseListItem;