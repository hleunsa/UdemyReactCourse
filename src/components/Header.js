import React from 'react'
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../firebase/firebase';

export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
        <h2><NavLink to="/" activeClassName="is-active" exact={true} >Home</NavLink></h2>
        <h2><NavLink to="/create" activeClassName="is-active">Create</NavLink></h2>
        <h2><NavLink to="/help" activeClassName="is-active">Help</NavLink></h2>
        <button onClick={startLogout}>logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);