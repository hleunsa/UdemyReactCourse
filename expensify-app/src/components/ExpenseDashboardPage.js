
import React from 'react';

import ExpenseList from './ExpenseList';

const ExpenseDashboardPage = () => (
    <div>
        <h1>This from my dashboard </h1>
        <ExpenseList/>
    </div>
);

export default ExpenseDashboardPage;