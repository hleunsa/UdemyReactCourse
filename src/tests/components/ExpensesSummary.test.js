import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';

test ('should correctly render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={425}/>);
    expect(wrapper).toMatchSnapshot();
});

test ('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={560}/>);
    expect(wrapper).toMatchSnapshot();
});