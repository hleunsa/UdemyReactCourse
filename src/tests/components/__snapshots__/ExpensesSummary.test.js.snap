// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $5.60
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $4.25
  </h1>
</div>
`;
